data/users/test/test_exceptions.py

```diff
@@ -0,0 +1,273 @@
+            admin_passwd="stillinvalid",
+            user_rdn="ou=people,dc=example,dc=com",
+            uid_attr="uid",
+            email_attr="mail",
+            timeout=1,
+            network_timeout=1,
+        )
+
+        def raiseException(ex):
+            raise ex(*[{"info": "test", "desc": "test"}])
+
+        user._ldap.get_connection = lambda: raiseException(exception[0])
+        assert user.ping() == (False, f"{exception[0].__name__} {exception[0].errnum} test")
+        assert exceptHandler.content[0] == "Exception when trying to health check LDAP"
+        assert exceptHandler.content[1] == f"{exception[0].__name__} {exception[0].errnum} test"
+        # reset content from log
+        exceptHandler.content = []  # type: List[str]
+
+
+def test_ldap_warning_exception_at_least_one_user_exists():
+    # reset content from log
+    exceptHandler.content = []  # type: List[str]
+
+    for exception in (
+        (ldap.AUTH_UNKNOWN, -6),
+        (ldap.CONFIDENTIALITY_REQUIRED, 13),
+        (ldap.CONSTRAINT_VIOLATION, 19),
+        (ldap.INVALID_DN_SYNTAX, 34),
+        (ldap.INVALID_SYNTAX, 21),
+        (ldap.SERVER_DOWN, -1),
+        (ldap.STRONG_AUTH_NOT_SUPPORTED, 7),
+        (ldap.STRONG_AUTH_REQUIRED, 8),
+        (ldap.UNAVAILABLE, 52),
+        (ldap.UNWILLING_TO_PERFORM, 53),
+        (ldap.CONTROL_NOT_FOUND, -13),
+        (ldap.INAPPROPRIATE_AUTH, 48),
+        (ldap.INSUFFICIENT_ACCESS, 50),
+    ):
+        user = LDAPUsers(
+            ldap_uri="ldaps://localhost",
+            base_dn=["dc=example", "dc=com"],
+            admin_dn="cn=invalid,dc=example,dc=com",
+            admin_passwd="stillinvalid",
+            user_rdn="ou=people,dc=example,dc=com",
+            uid_attr="uid",
+            email_attr="mail",
+            timeout=1,
+            network_timeout=1,
+        )
+
+        def raiseException(ex):
+            raise ex(*[{"info": "test", "desc": "test"}])
+
+        user._ldap.get_connection = lambda: raiseException(exception[0])
+        assert user.at_least_one_user_exists() == (None, "LDAP Admin dn or password is invalid")
+        assert exceptHandler.content[0] == f"{exception[0].__name__} {exception[1]} test"
+        # reset content from log
+        exceptHandler.content = []  # type: List[str]
+
+
+def test_ldap_warning_exception_verify_credentials():
+    # reset content from log
+    exceptHandler.content = []  # type: List[str]
+
+    for exception in (
+        (ldap.AUTH_UNKNOWN, -6),
+        (ldap.CONFIDENTIALITY_REQUIRED, 13),
+        (ldap.CONSTRAINT_VIOLATION, 19),
+        (ldap.INVALID_DN_SYNTAX, 34),
+        (ldap.INVALID_SYNTAX, 21),
+        (ldap.SERVER_DOWN, -1),
+        (ldap.STRONG_AUTH_NOT_SUPPORTED, 7),
+        (ldap.STRONG_AUTH_REQUIRED, 8),
+        (ldap.UNAVAILABLE, 52),
+        (ldap.UNWILLING_TO_PERFORM, 53),
+        (ldap.CONTROL_NOT_FOUND, -13),
+        (ldap.INAPPROPRIATE_AUTH, 48),
+        (ldap.INSUFFICIENT_ACCESS, 50),
+    ):
```

Comment:
😀 we are duplicating this in all the tests, if we have to add or remove an exception it would require updating in every place. Best if we have one source of truth for the exceptions